src/serveractiveobjectmap.h

```diff
@@ -0,0 +1,67 @@
+/*
+Minetest
+Copyright (C) 2010-2017 numZero, Lobachevsky Vitaly <numzer0@yandex.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+#include <unordered_set>
+#include <unordered_map>
+#include <vector>
+#include "irr_v3d.h"
+#include "irr_aabb3d.h"
+
+class ServerActiveObject;
+
+struct ServerActiveObjectMap
+{
+	struct Wrapper
+	{
+		ServerActiveObject *object;
+		aabb3s16 box;
+		bool has_box;
+	};
+
+	void addObject(ServerActiveObject *object);
+	void removeObject(ServerActiveObject *object);
+	ServerActiveObject *removeObject(u16 id);
+	void updateObject(ServerActiveObject *object);
+	void updateObject(u16 id);
+	ServerActiveObject *getObject(u16 id) const;
+
+	bool isFreeId(u16 id);
+	u16 getFreeId();
+	std::vector<u16> getObjectsInsideRadius(v3f pos, float radius);
```

Comment:
please pass in a std::vector<u16>&

Reply:
@rubenwardy Why? IMO the only reason for that might be to let this function append to a vector instead of creating a new one. Is that what you meant?

Reply:
Note that in the current code, the vector is *moved* and never copied. That’s C++11.

Reply:
you didn't use move operator i don't see why compiler will do it

Reply:
C++ standard specifies that compilers will optimize returning an lvalue to moving it.

Reply:
Just researched this, and yes it will be optimised to a `move()`